### PR TITLE
Fixing code

### PR DESCRIPTION
**Pull request by OPpuolitaival**
Small fixing like: 
- Initialise variable before usage
- Match string arguments amount
- Remove unused variables

### CODE REVIEW

**Comment by OPpuolitaival** [comment]
@theotherjimmy I will fix that problem

File: mbed/mbed.py [hunk 1]
```diff
@@ -1278,6 +1279,7 @@ def get_cache(self, url):
     def set_cache(self, url):
         up = urlparse(formaturl(url, 'https'))
         if self.cache and up and up.netloc and os.path.isdir(self.path):
+            cpath = 'none'
```
Comment (screamerbg) [inline mbed/mbed.py]: This is not right. `cpath = os.path.join(...` should be moved before the `try:` instead

**Comment by OPpuolitaival** [comment]
Fixed @screamerbg mentioned problem and added auto formatting

**Comment by screamerbg** [comment]
@OPpuolitaival LGTM!

**Comment by screamerbg** [comment]
@OPpuolitaival Could you please rebase?

**Comment by OPpuolitaival** [comment]
@screamerbg rebased

**Comment by screamerbg** [comment]
@OPpuolitaival Can you remove the autoformatting? That's way too intrusive and I don't agree with some of the autoformatting rules.

**Comment by screamerbg** [comment]
@OPpuolitaival I mean whatever was introduced in https://github.com/ARMmbed/mbed-cli/pull/640/commits/85829a210cf4b61ddad478e33c7ef2b8c4a5fa5e

**Comment by OPpuolitaival** [comment]
@screamerbg autoformatting removed. I did ask from Jimmy and he was okay with those and therefore added it in here

**Comment by screamerbg** [comment]
@OPpuolitaival Thanks. Current change set is 15 changed lines vs 580 with autoformatter.

**Comment by OPpuolitaival** [comment]
@screamerbg Sure, that is the idea of autoformatting that it does big job automatically

**Pull request merged**